### DISCUSSION TITLE
Niftisaver doesn't save new voxelspacing

### DISCUSSION THREAD

**Discussion by Gijz33** [Project-MONAI/MONAI#3299] [Q&A] [answered]
Hi,

I am changing the voxel spacing of my CT data using monai.transforms.Spacingd and saving the CT with new spacing into a .nii.gz-file using monai.data.NiftiSaver.
The voxel size transformation is succesful inside the python script. However, opening the .nii.gz file using different software (3D-slicer) the spacing of the old/orginal CT is used. Can someone help me out? A snippet of my code is shown below:

```python
train_images = sorted(glob.glob(os.path.join(data_folder, "*_ct.nii.gz")))
train_labels = sorted(glob.glob(os.path.join(data_folder, "*_seg.nii.gz")))

data_dicts = [    {"image": image_name, "label": label_name} for image_name, label_name in zip(train_images, train_labels)    ]

loader = LoadImaged(keys=("image","label"))
data_dict = loader(data_dicts[0])
     
add_channel = AddChanneld(keys=["image", "label"])
data_dict = add_channel(data_dict)

orientation = Orientationd(keys=["image", "label"], axcodes="LPS")
data_dict = orientation(data_dict)

spacing = Spacingd(keys=["image", "label"], pixdim=(0.8, 0.8, 3.0), mode=("bilinear")) 
data_dict = spacing(data_dict)


saver = NiftiSaver(output_dir="./", output_postfix="test" ,output_ext=".nii.gz",mode="nearest")
saver.save(data_dict["image"], data_dict['image_meta_dict'])
```

**Comment by wyli** [comment, marked as answer]
by default the NiftiSaver will resume the original space, if you don't want this behaviour, you can remove the original info:
```py
data_dict["image_meta_dict"]["original_affine"] = None
saver = NiftiSaver(output_dir="./", output_postfix="test", output_ext=".nii.gz", mode="nearest")
saver.save(data_dict["image"], data_dict["image_meta_dict"])    
```

**Reply by Gijz33** [reply]
Thank you for your help @wyli!
It does indeed store the correct voxel spacing now. However, I got the feeling that more than just the voxel spacing has changed.
Because the image is now very different from the original. Is is possible that more parameters have been changed?

Update: I'm sorry, it is working perfectly fine, I made a mistake. Thanks again for your help!

**Reply by wyli** [reply]
thanks for the feedback!
